packages/autorest.python/autorest/codegen/models/model_type.py

```diff
@@ -79,18 +81,33 @@ def serialization_type(self) -> str:
             )
         return "object"
 
+    @property
+    def _is_polymorphic(self) -> bool:
+        return any(p.is_polymorphic for p in self.properties)
+
     def type_annotation(self, **kwargs: Any) -> str:
         if self.code_model.options["models_mode"]:
             is_operation_file = kwargs.pop("is_operation_file", False)
             retval = f"_models.{self.name}"
             if not self.is_public:
                 retval = f"{self.code_model.models_filename}.{retval}"
+            if self.code_model.options["models_mode"] == "dpg" and kwargs.pop(
+                "is_body_parameter", False
+            ):
+                retval += ", JSON"
+                if not kwargs.pop("in_combined_type", False):
```

Comment:
I think we can leverage combined_type.py's type_annotation instead of passing whether a kwarg saying `in_combined_type`. I.e. we can have the `type_annotation` in combined_type.py add `Union[{retval}]`

Reply:
Here it's for assigning model annotation as `Union[MyModel, JSON]` when the model is NOT belong to a CombinedType, so the combined_type.py may not help :).

And I need this not-in-combined checking because we need `Union[MyModel, JSON, IO]`  instead of `Union[Union[MyModel, JSON], IO]` when the model type is combined with a `IO` type.

Reply:
actually @changlong-liu I still think this should be in `combined_type.py`. You can do a check there and remove the `Union` if we don't want duplicates, but I think it's still cleaner to keep the type annotation for combined type within combined type. That way if we have an issue with type_annotation for a type, we know exactly where to go. We can leave a comment to about what we're doing

Reply:
removed `in_combined_type` and adding logic in CombinedType to avoid duplicated `Union`.